src/App.vue

```diff
@@ -369,127 +392,138 @@ const vehicleStore = useMainVehicleStore()
+    component: markRaw(ConfigurationActionsView) as SubMenuComponent,
+  },
+]
+
+const toolsMenu = [
+  {
+    icon: 'mdi-protocol',
+    title: 'MAVLink',
+    component: markRaw(ToolsMAVLinkView) as SubMenuComponent,
+  },
+  {
+    icon: 'mdi-database-outline',
+    title: 'Data-lake',
+    component: markRaw(ToolsDataLakeView) as SubMenuComponent,
   },
 ]
 
+const availableSubMenus = {
+  config: configMenu,
+  tools: toolsMenu,
+}
+
+const currentSubMenu = computed(() => {
+  if (currentSubMenuName.value === null) return []
+  return availableSubMenus[currentSubMenuName.value]
+})
+
 watch(
-  () => interfaceStore.configComponent,
-  (component) => {
-    if (component < 0) {
-      currentConfigMenuComponent.value = null
-      return
-    }
-    currentConfigMenuComponent.value = configMenu[component].component
+  () => interfaceStore.subMenuComponentName,
+  (subMenuComponentName) => {
+    currentSubMenuComponent.value =
+      currentSubMenu.value.find((item) => item.title === subMenuComponentName)?.component || null
   }
 )
 
-const toggleConfigComponent = (component: ConfigComponent): void => {
-  if (currentConfigMenuComponent.value === null) {
-    currentConfigMenuComponent.value = component
+const selectSubMenu = (subMenuName: keyof typeof availableSubMenus): void => {
+  currentSubMenuName.value = subMenuName
+  interfaceStore.mainMenuCurrentStep = 2
+}
+
+const toggleSubMenuComponent = (component: SubMenuComponent): void => {
+  if (currentSubMenuComponent.value === null) {
+    currentSubMenuComponent.value = component
     interfaceStore.configModalVisibility = true
     return
   }
-  if (currentConfigMenuComponent.value === component) {
-    currentConfigMenuComponent.value = null
+  if (currentSubMenuComponent.value === component) {
+    currentSubMenuComponent.value = null
     interfaceStore.configModalVisibility = false
     return
   }
-  currentConfigMenuComponent.value = component
+  currentSubMenuComponent.value = component
   interfaceStore.configModalVisibility = true
 }
 
 const isConfigModalVisible = computed(() => interfaceStore.isConfigModalVisible)
 
 watch(isConfigModalVisible, (newVal) => {
   if (newVal === false) {
-    currentConfigMenuComponent.value = null
+    currentSubMenuComponent.value = null
   }
 })
 
-watch(
-  () => windowHeight.value < 450,
-  (isSmall: boolean) => {
-    simplifiedMainMenu.value = isSmall
-  }
-)
-
-const updateWindowHeight = (): void => {
-  windowHeight.value = window.innerHeight
-}
+const topBottomBarScale = computed(() => {
+  return windowWidth.value / originalBarWidth
+})
 
-onMounted(() => {
-  window.addEventListener('resize', updateWindowHeight)
-  if (windowHeight.value < 450) {
-    simplifiedMainMenu.value = true
-  }
+const maxScreenHeightPixelsThatFitsLargeMenu = computed(() => {
+  const heightTopBar = widgetStore.currentTopBarHeightPixels * topBottomBarScale.value
+  const heightBottomBar = widgetStore.currentBottomBarHeightPixels * topBottomBarScale.value
```

Comment:
Given how the sidebar is changing, I think these might not be being calculated correctly, and I'm still not sure why they're based on `windowWidth`